### ISSUE TITLE
Android app crashes when editing notes over 100,000 chars

### ISSUE THREAD

**Issue by mihaiburlacu** [standardnotes/forum#994]
Android app crashes when editing notes over 100,000 chars

1. Go to Android Standard Notes app
2. Click on any note larger than 100,000 characters
3. Try entering any character (editing the note content in any way)
4. Android app crashes.
Please note app does not crash when editing only the title of the note.

**Expected behavior**
Android app does not crash.

**Smartphone**
 - Device: Motorola One Action
 - OS: Android 10
 - Version: Standard Notes 3.0.21

**Comment by JaspalSuri** [comment]
Hi, I'm sorry that you're experiencing issues with this.

I wasn't able to reproduce this issue on a Pixel 3a. I looked up the specs of the Motorola One Action, and it seems like it should be able to handle large files. 

Thanks for all of the information that you've provided! Can you reproduce this issue with any other text editor app? If you don't want to paste over your private notes, you can use [this site](http://www.unit-conversion.info/texttools/random-string-generator/) to generate the 100k characters instead.

@radko93, can you reproduce this on your end?

**Comment by johnny243** [comment]
I could not reproduce this issue. Although, the app crashed on me when trying to paste text with over +800,000 chars (~715 Kb).

**Comment by mihaiburlacu** [comment]
Hi, 

I could not reproduce this issue with another text editor: File Manager+'s internal text editor and QuickEdit Pro handle ~106,000 characters well, copying to clipboard also works, whereas Standard Notes crashes. 

When raising this topic I did a test with ~96k characters that Standard Notes handled just fine (that's how I assumed the limit to be about 100k chars), however now 90k chars lead to a crash (80k chars work just fine).

Past 800,000 chars only QuickEdit seemed to work, while other text editors lag severely and copying to clipboard is not possible.

**Comment by mihaiburlacu** [comment]
Since this is not reproducible on other devices and the limit seems to vary even on my device, I can accept that issue might be localized/device-related somehow.

**Comment by JaspalSuri** [comment]
Thanks for the additional information! I will be sure to pass this along for review.

**Comment by pflan** [comment]
I've also had some crashes while editing large notes, but inconsistently: I can't reliably reproduce.

Today the app crashed while editing a much smaller note.

I'm on version 3.5.14, Android 11, Pixel 3 XL.

**Comment by JaspalSuri** [comment]
Hi @pflan,

Thanks for letting us know; I'll be sure to forward this to the dev team. 

Are you using a dark theme on the Android app along with the system dark theme?

Can you try signing into our [legacy web app](https://app-legacy.standardnotes.org) and let us know if that works? If not, please sign into the latest [stable web app](https://app.standardnotes.org), click on **Options**, and sort your notes by **Date Modified** to ensure that your changes have been properly saved to the server (if you're using an account).

Have you already tried clearing the app's storage and cache? 

Looking forward to hearing back from you!

**Comment by pflan** [comment]
I am using the Standard Notes dark theme, but not the system dark theme.

I also have the Windows app and the Linux app and I haven't experienced crashes with them.

I had not tried clearing the app's storage and cache, no. I just tried signing out and signing back in again, we'll see if that helps. (I tried to reproduce just now, but as I mentioned, I can't reliably reproduce a crash.)

I have no experience with either the legacy web app or the stable web app, but I just tried the legacy web app. It sort of worked, in that I was able to log in, but the notes editor panel was stuttering (as though a CSS element was being hidden and then revealed every 200ms), and then after about 1 minute I started getting "403 Forbidden" errors. It appears that my IP address is now blocked on the website side, because I get "403 Forbidden" even when visiting www.standardnotes.org (i.e. not the web app) on my Windows PC (i.e. not my mobile device, same network), but I do not get the same error when visiting the website through a proxy (ProtonVPN). ::shrug:: This is probably a red herring. (The app still works fine, on all devices, Android, Windows, and Linux, it's just the web app and now the basic website that I'm getting "403 Forbidden" errors from.)
